cli/internal/cmd/upgrade.go

```diff
@@ -23,3 +27,24 @@ func NewUpgradeCmd() *cobra.Command {
 	cmd.AddCommand(newUpgradeApplyCmd())
 	return cmd
 }
+
+// upgradeCmdKind is the kind of the upgrade command (check, apply).
+type upgradeCmdKind int
+
+const (
+	// upgradeCmdKindCheck corresponds to the upgrade check command.
+	upgradeCmdKindCheck upgradeCmdKind = iota
+	// upgradeCmdKindApply corresponds to the upgrade apply command.
+	upgradeCmdKindApply
+)
+
+func generateUpgradeID(kind upgradeCmdKind) string {
```

Comment:
nit: should iam upgrade use this too?

Reply:
I applied this suggestion, but this also means IAM upgrades are placed in folders similar named to other upgrades.
Is this something we are OK with, or should be adjust this to stay as `iam-<uid>` ?

Reply:
Added a new `upgradeCmdKind` for IAM